controllers/admin/AdminModulesController.php

```diff
@@ -1438,10 +1439,14 @@ public function initContent()
         // If we are on a module configuration, no need to load all modules
         if (Tools::getValue('configure') != '') {
             $this->context->smarty->assign(array('maintenance_mode' => !(bool)Configuration::Get('PS_SHOP_ENABLE')));
-
             return true;
         }
 
+        // Since 1.7, legacy modules page does not have to be show
+        // Redirect to the new page, can not do this into __construct because
+        // some install module, configuration are done with this controller...
+        Tools::redirectAdmin($this->getAdminModulesUrl());
```

Comment:
If we force the redirection here, we can remove the next lines until the end of the function.